app/uk/gov/hmrc/selfassessmentapi/repositories/domain/MongoLiability.scala

```
@@ -44,7 +45,8 @@ case class MongoLiability(id: BSONObjectID,
                           savingsIncome: Seq[TaxBandAllocation] = Nil,
                           dividendsIncome: Seq[TaxBandAllocation] = Nil,
                           allowancesAndReliefs: AllowancesAndReliefs = AllowancesAndReliefs(),
-                          taxDeducted: Option[MongoTaxDeducted] = None) extends Math {
+                          taxDeducted: Option[MongoTaxDeducted] = None,
+                          calculationError: Option[CalculationError] = None) extends Math {
```

Should this be singular? The calculation should fail fast if an error occurs, but there could be multiple causes of the error. In the employments tax paid example, several sources could have a tax paid <0, so if we wanted to report to the consumer that particular sources were in an invalid state, how would we do that?